src/sparsezoo/v2/model_directory.py

```diff
@@ -230,55 +239,37 @@ def download(self, directory_path: str, override: bool = False) -> bool:
 
         return all(downloads)
 
-    def validate(self) -> bool:
+    def validate(
+        self, validate_onnxruntime: bool = True, minimal_validation: bool = False
+    ) -> bool:
         """
         Validate the ModelDirectory class object:
         1. Validate that the sample inputs and outputs work with ONNX Runtime
-        2. Validate all the folders
-
+            (if `validate_onnxruntime=True`)
+        2. Validate all the folders (this is done by a separate helper class
+            IntegrationValidator)
+
+        :param validate_onnxruntime: boolean flag; if True, validate that the
+            sample inputs and outputs work with ONNX Runtime
+        :param minimal_validation: boolean flag; if True, only the essential files
+            in the `training` folder are validated. Else, the `training` folder is
+            expected to contain a full set of framework files.
         return: a boolean flag; if True, the validation has been successful
         """
 
-        if not self.inference_runner.validate_with_onnx_runtime():
-            logging.warning(
-                "Failed to validate the compatibility of "
-                "`sample_inputs` files with the `model.onnx` model."
-            )
-            return False
-
-        # TODO: This is a hack for now,
-        #  some files cannot be validated
-        #  using dummy inputs (see respective tests)
-        SKIP_ATTRIBUTES = ["training"]
-
-        if self.path is None:
-            raise ValueError(
-                "Cannot validate the ModelDirectory. "
-                "If created using method `from_directory`, "
-                "please make sure that the `directory_path` is correct. "
-                "If created using method `from_zoo_api`, "
-                "call `download()` method prior to `validate()`"
-            )
+        if validate_onnxruntime:
+            if not self.inference_runner.validate_with_onnx_runtime():
+                logging.warning(
+                    "Failed to validate the compatibility of "
+                    "`sample_inputs` files with the `model.onnx` model."
+                )
+                return False
```

Comment:
Good point. This is actually an interesting engineering question. We might be raising warnings (that do not interrupt the code), or value errors (that do stop the script) or have a flag (e.g. `strict_mode = False`), that would allow to choose between both.
@bfineran do you have a opinion on this? Should failed validation just be flagged using logging warnings and the function should gracefully return `False`, or would we rather like to be strict and raise value errors anytime something goes south?

Reply:
Good point. This is actually an interesting engineering question. We might be raising warnings (that do not interrupt the code), or value errors (that do stop the script) or have a flag (e.g. strict_mode = False), that would allow to choose between both.
@bfineran do you have a opinion on this? Should failed validation just be flagged using logging warnings and the function should gracefully return False, or would we rather like to be strict and raise value errors anytime something goes south?